commit 83250e67a9d65a662c5fdec6075f3c2dee0e79a0
Author: Jerry Hoemann <jerry.hoemann@hpe.com>
Date:   Wed Jan 6 16:03:41 2016 -0700

    libnvdimm: Fix security issue with DSM IOCTL.
    
    commit 07accfa9d1a8bac8262f6d24a94a54d2d1f35149 upstream.
    
    Code attempts to prevent certain IOCTL DSM from being called
    when device is opened read only.  This security feature can
    be trivially overcome by changing the size portion of the
    ioctl_command which isn't used.
    
    Check only the _IOC_NR (i.e. the command).
    
    Signed-off-by: Jerry Hoemann <jerry.hoemann@hpe.com>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/nvdimm/bus.c b/drivers/nvdimm/bus.c
index 7e2c43f..496b9b6 100644
--- a/drivers/nvdimm/bus.c
+++ b/drivers/nvdimm/bus.c
@@ -513,10 +513,10 @@ static int __nd_ioctl(struct nvdimm_bus *nvdimm_bus, struct nvdimm *nvdimm,
 
 	/* fail write commands (when read-only) */
 	if (read_only)
-		switch (ioctl_cmd) {
-		case ND_IOCTL_VENDOR:
-		case ND_IOCTL_SET_CONFIG_DATA:
-		case ND_IOCTL_ARS_START:
+		switch (cmd) {
+		case ND_CMD_VENDOR:
+		case ND_CMD_SET_CONFIG_DATA:
+		case ND_CMD_ARS_START:
 			dev_dbg(&nvdimm_bus->dev, "'%s' command while read-only.\n",
 					nvdimm ? nvdimm_cmd_name(cmd)
 					: nvdimm_bus_cmd_name(cmd));
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1284_83250e67a9d65a662c5fdec6075f3c2dee0e79a0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1284_83250e67a9d65a662c5fdec6075f3c2dee0e79a0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
